app/src/main/java/ch/epfl/sdp/blindly/helpers/BlindlyLatLng.kt

```diff
@@ -0,0 +1,44 @@
+package ch.epfl.sdp.blindly.helpers
+
+import android.location.Location
+import com.google.android.gms.maps.model.LatLng
+
+
+// Simple wrapper to let firebase handle it nicely
+class BlindlyLatLng {
+    private var latitude: Double? = null
+    private var longitude: Double? = null
+
+    @Suppress("unused") // needed for firebase
+    fun getLatitude(): Double? {
+        return latitude
+    }
+
+    @Suppress("unused") // needed for firebase
+    fun getLongitude(): Double? {
+        return longitude
+    }
+
+    @Suppress("unused") // needed for firebase
+    constructor() {}
+
+    constructor(latitude: Double?, longitude: Double?) {
+        this.latitude = latitude
+        this.longitude = longitude
+    }
+    constructor(latLng: LatLng) {
+        latitude = latLng.latitude
+        longitude = latLng.longitude
+    }
+
+    constructor(latLng: Location?) {
+        this.latitude = latLng?.latitude
+        this.longitude = latLng?.longitude
+    }
+
+    fun toLatLng(): LatLng? {
+        latitude ?: return null
+        longitude ?: return null
+        return LatLng(latitude!!, longitude!!)
+    }
+}
```

Comment:
To be honest I don't really understand the purpose of this class. Isn't it exactly the same as a LatLng object that already exists in Kotlin ?

Reply:
It is very similar indeed, however Firebase had troubles receiving a LatLng objects so I wrapped it. Moreover I had a lot of code that needed Lat/Lng data in different formats (Location, LatLng, Doubles) so having a single class do everything avoid some boilerplate code. 
Furthermore the LatLng comes from the Google API, I think it is arguably better to wrap external classes that we depend on, especially given the stability of the APIs on android.

Reply:
Ok, thanks for the explanation !